rmgpy/rmg/model.py

```diff
@@ -674,22 +693,81 @@ def enlarge(self, newObject=None, reactEdge=False,
+
+            # Get new species and save in spcs
+            spcs_tmp = []
+            for rxn in rxns:
+                spcs_tmp.extend(rxn.reactants)
+                spcs_tmp.extend(rxn.products)
+                
+            spcs = spcs_tmp
+
+            from rmgpy.rmg.input import getInput
+            try:
+                quantumMechanics = getInput('quantumMechanics')
+            except Exception:
+                logging.debug('Quantum Mechanics DB could not be found.')
+                quantumMechanics = None
+        
+            if not quantumMechanics:
+                pass
+            else:
+                if not spcs_tmp:
+                    spcs = spcs_tmp
+                else:
+                    # Generate unique list of species to be submitted to QM thermo calculation
+                    # intilize list 
+                    spcs=[spcs_tmp[0]]
+                    for counter, spc in enumerate (spcs_tmp):
+                        for counter2, val in enumerate (spcs):
+                            if (spc.molecule[0].toSMILES() != val.molecule[0].toSMILES()):
+                                appendReactant = True
+                            else:
+                                appendReactant = False
+                                break
+                        if appendReactant:
+                            spcs.append(spc)
+    
+                    # Calculate quantum thermo in parallel
+                    from rmgpy.rmg.main import maxproc
+                    
+                    # Get available RAM (GB)and procnum dependent on OS
+                    if platform.startswith('linux'):
+                        # linux
+                        memoryavailable = psutil.virtual_memory().free / (1000.0 ** 3)
+                        memoryuse = psutil.Process(os.getpid()).memory_info()[0]/(1000.0 ** 3)
+                        tmp = divmod(memoryavailable, memoryuse)
+                        tmp2 = min(maxproc, tmp[0])
+                        procnum = max(1, int(tmp2))
+                    elif platform == "darwin":
+                        # OS X
+                        memoryavailable = psutil.virtual_memory().available/(1000.0 ** 3)
+                        memoryuse = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss/(1000.0 ** 3)
+                        tmp = divmod(memoryavailable, memoryuse)
+                        tmp2 = min(maxproc, tmp[0])
+                        procnum = max(1, int(tmp2))
+                    else:
+                        # Everything else
+                        procnum = 1
+                
+                    # Execute multiprocessing map. It blocks until the result is ready.
+                    # This method chops the iterable into a number of chunks which it
+                    # submits to the process pool as separate tasks.
+                    p = Pool(processes=procnum)
+                    p.map(calculate_thermo_parallel,spcs)
```

Comment:
I have a couple concerns about the overall implementation for parallel QM. Normally, thermo is calculated for one species at a time in `self.processNewReactions()` below. Here, you determine all species which would be calculated using QM and pre-calculate their thermo in parallel, then in `processNewReactions`, it sees the thermo and does not re-calculate it.

1. The `calculate_thermo_parallel` function seems to skip a lot of processing done in `ThermoDatabase.getThermoData` and `thermoEngine.processThermoData`, which can have a significant effect on the final values.
2. It appears that you're using SMILES to determine whether species are unique? This is not safe because different resonance structures will have different SMILES. Also, generating SMILES for every new molecule will be relatively time consuming.

Minor suggestions, which may or may not be relevant after resolving the above issues:
- Since the procnum determination based on RAM is done both here and in react.py, it might be useful to turn it into a function somewhere and import it here.
- I would also split this section off as a new method, since `enlarge` is already a really big method.

Reply:
The thought behind the implemented procedure is that the QM files should be generated in parallel and then looked up for one species at a time in self.processNewReactions() below. Testing showed that calculating QM thermo for one species at a time is more time consuming than in a bulk up here. 

I use SMILES for comparison and then generate resonance structures within calculate_thermo_parallel() to generate QM files for each resonance structure. What would be a better way to ensure that the same QM file is not generated twice/overlapping during the parallel generation?

Reply:
The SMILES comparison is now substituted with isIsomorphic comparison. 

The procnum determination is now a function and imported for QMTP parallel and reaction generation in parallel.

And the parallel QM files writing section is split off outside of enlarge.